### PR TITLE
Pipeline lineage edge UI

### LINKED ISSUE #5653: UI - Handle Pipeline Lineage as an Edge Transformation
Based on the ingestion changes on https://github.com/open-metadata/OpenMetadata/issues/5444

We should support showing the `pipeline` information from lineage details as a pipeline edge. This change was breaking the UI and we rolled back the ingestion changes. Let us know when you'll work on this so that we can add them back.

This was the reverted change https://github.com/open-metadata/OpenMetadata/pull/5552

cc @OnkarVO7

### PR DESCRIPTION
**Pull request by OnkarVO7**
### Describe your changes :

Fix https://github.com/open-metadata/OpenMetadata/issues/5572
Closes #5653

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/71748675/179210174-1b032154-14b1-4c38-8dfb-4ad9cb8f4d1f.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@open-metadata/ingestion @open-metadata/ui @open-metadata/backend

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/AddPipeLineModal.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        </Button>
+      </div>
+    );
+  };
+
+  return (
+    <Modal
+      destroyOnClose
+      data-testid="add-pipeline-modal"
+      footer={Footer()}
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/AddPipeLineModal.tsx:76]:
```suggestion
      footer={<Footer/>}
```

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomEdge.component.tsx [hunk 2]
```diff
@@ -30,7 +34,7 @@ export const CustomEdge = ({
   data,
   selected,
 }: EdgeProps) => {
-  const { onEdgeClick, ...rest } = data;
+  const { onEdgeClick, addPipelineClick, ...rest } = data;
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomEdge.component.tsx:37]:
```suggestion
  const { onEdgeClick, addPipeline, ...rest } = data;
```

Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomEdge.component.tsx:37]: I think this should be fine as I am indicating click for add pipeline button

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomEdge.component.tsx:37]: That is the callback name, so I think we don't need to name it with the `Click` suffix.

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomEdge.component.tsx:37]: Got it.

**Review by Sachin-chaurasiya — APPROVED** [review]

**Review by pmbrull — APPROVED** [review]

**Pull request merged**